### ISSUE TITLE
Bug: not checking if function exists when containing variables as part of the call.

### ISSUE THREAD

**Issue by nihilus** [koalaman/shellcheck#1079]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x ] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [x ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/your/interpreter
foo(){
...
}

foo"$bar"
```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:
It should yield an error since eiether I want to call 'foo "$bar"' or 'foo$bar'.
